workers/gossip-server/src/peer_discovery/peers.rs

```diff
@@ -159,6 +150,11 @@ impl GossipProtocolExecutor {
                 continue;
             }
 
+            // Check that the peer is not already in the peer index
+            if self.state.get_peer_info(&peer.peer_id).await?.is_some() {
+                continue;
+            }
```

Comment:
why do we want to do this? and if we do, I'd suggest moving this to `state`, if this solves a bug it should be solved in the `state` interface

Reply:
we do this because the re-addition of a peer overwrites our local view of the peer's last heartbeat with whatever was sent in the payload here. this can lead to us writing in stale heartbeat timestamps, as opposed to the local node's actual view.

I did this here as opposed to `state.add_peers_batch` since nodes filtered according to this clause shouldn't have their addresses written in via `add_new_addrs`, but now that I mention it, we should probably decouple these two notions. i.e. avoid overwriting heartbeats for a node we've already indexed to maintain an accurate view of liveness, but allow the other metadata to be updated.

Will do this in `state`